test/m.jl

```diff
@@ -0,0 +1,9 @@
+#= mxvg is Severely Broken, test below does not work and dont know why!
+ Perform the following calculation: 1*1 + 2*2 + 1*3, 3*1 + 1*2 + 4*3
+@testset "M" begin 
+    matrix = [1. 1. 1. 
+                2. 3. 4.]
+    vecgood = [1., 2., 3.]
+    @test mxvg(matrix, vecgood) == [6.,20.]
```

Comment:
The problem is that CSPICE expects the matrix to be in row-major order whereas Julia uses column-major order.
We need to pass the transpose of the input matrix to the C code, e.g.:
```julia
function mxvg(m1, v2)
    lm1, lm2 = size(m1)
    lv = length(v2)
    if lm2 != lv
        error("Dimension mismatch.")
    end
    vout = Array{Float64}(lm1)
    ccall((:mxvg_c, libcspice), Void, (Ptr{Float64}, Ptr{Float64}, Cint, Cint, Ptr{Float64}),
    #= use transpose =# m1', v2, lm1, lm2, vout)
    handleerror()
    return vout
end
```